### PR TITLE
TASK: Adjust responsibilities for validation result rendering

### LINKED ISSUE #1769: Some Editors does not render validation errors
### Description

If you create a custom createDialog like in issue #1320 you will recognize that some editors will not render the validation errors. It seams that only the textInput and textArea actually respect the validationError property and render the tooltip.

The DateTimeInput has been adjusted already in issue #1320 ... so probably all others excerpt the
three will have this validation error rendering issue.

### Steps to Reproduce

1. Configure a createDialog and add some additional fields (see the 1320 issue)
2. Create a new page without entering stuff

#### Expected behavior

For all possible editors the validation errors should be rendered. The validationErrors prop will exist for all on a higher level. The EditorEnvelope pass thru the props for all.

#### Actual behavior

Validation will be done, but the user gets no visual feedback for some field types.

[What actually happened]

### Affected Versions

Neos: 3.3

UI: 1.0.11 and lower

### PR DESCRIPTION
**Pull request by johannessteu**
This change takes moves the validation result rendering to a central place into EditorEnvelope. Also core ui components like a select box or a textfield should not care about stuff like `isHighlighted` oder rendering validation results. Those options are removed and will be handled from `EditorEnvelope`

Resolves #1769  #1651 #1426 (as we render tooltip with validation inline)

### CODE REVIEW

**Comment by johannessteu** [comment]
Resolves #1769  #1651 #1426 (as we render tooltip with validation inline)

**Comment by johannessteu** [comment]
Some examples with faked validation errors / highlighted styles

bildschirmfoto 2018-06-28 um 21 32 17
bildschirmfoto 2018-06-28 um 21 32 08
bildschirmfoto 2018-06-28 um 21 32 13
bildschirmfoto 2018-06-28 um 21 31 23
bildschirmfoto 2018-06-28 um 21 31 15

File: packages/neos-ui-editors/src/EditorEnvelope/index.js [hunk 1]
```diff
@@ -54,15 +60,22 @@ export default class EditorEnvelope extends PureComponent {
...
             const EditorComponent = editorDefinition && editorDefinition.component;
 
+            const restProps = omit(this.props, ['validationErrors', 'highlight']);
+            const {highlight, validationErrors} = this.props;
+            const isInvalid = validationErrors && validationErrors.length > 1;
+
+            // We pass down a classname to render a highlight status on the editor field
+            const classNames = mergeClassNames({
+                [style['envelope--highlight']]: highlight && !isInvalid,
+                [style['envelope--invalid']]: validationErrors && validationErrors.length > 0
```
Comment (mstruebing) [inline packages/neos-ui-editors/src/EditorEnvelope/index.js]: Shouldn't this be the the same as `isInvalid` ?

File: packages/neos-ui-editors/src/EditorEnvelope/index.js [hunk 2]
```diff
@@ -54,15 +60,22 @@ export default class EditorEnvelope extends PureComponent {
         if (editorDefinition && editorDefinition.component) {
             const EditorComponent = editorDefinition && editorDefinition.component;
 
+            const restProps = omit(this.props, ['validationErrors', 'highlight']);
+            const {highlight, validationErrors} = this.props;
+            const isInvalid = validationErrors && validationErrors.length > 1;
```
Comment (mstruebing) [inline packages/neos-ui-editors/src/EditorEnvelope/index.js]: Shouldn't this be `> 0`? - as soon as one error occurs it is invalid I think

File: packages/neos-ui-editors/src/EditorEnvelope/index.js [hunk 3]
```diff
@@ -54,15 +60,22 @@ export default class EditorEnvelope extends PureComponent {
         if (editorDefinition && editorDefinition.component) {
             const EditorComponent = editorDefinition && editorDefinition.component;
 
+            const restProps = omit(this.props, ['validationErrors', 'highlight']);
+            const {highlight, validationErrors} = this.props;
```
Comment (mstruebing) [inline packages/neos-ui-editors/src/EditorEnvelope/index.js]: `const {highlight, validationErrors, ...restProps} = this .props`
That way you wouldn't need the `omit` function call

File: packages/neos-ui-editors/src/EditorEnvelope/index.js (hunk 2 above)
Comment (johannessteu) [reply packages/neos-ui-editors/src/EditorEnvelope/index.js]: fixed

File: packages/neos-ui-editors/src/EditorEnvelope/index.js (hunk 1 above)
Comment (johannessteu) [reply packages/neos-ui-editors/src/EditorEnvelope/index.js]: fixed

File: packages/neos-ui-editors/src/EditorEnvelope/index.js (hunk 3 above)
Comment (johannessteu) [reply packages/neos-ui-editors/src/EditorEnvelope/index.js]: fixed

**Pull request merged**